src/middlewares/error.middleware.ts

```diff
@@ -1,32 +1,78 @@
+import { DEV_ENV } from '@config';
+import { ErrorsMessages } from '@constants/errorMessages';
+import { HttpStatusCode } from '@constants/httpStatusCode';
+import { ValidationError } from 'class-validator';
 import { Request, Response } from 'express';
 import {
-  Middleware,
-  ExpressErrorMiddlewareInterface
+  ExpressErrorMiddlewareInterface,
+  Middleware
 } from 'routing-controllers';
-import { DEV_ENV } from '@config';
 import { Service } from 'typedi';
 
 interface ErrorInterface extends Error {
+  name: string;
   httpCode: number;
+  description: string | undefined;
+  errors: string[] | undefined;
 }
 
 @Middleware({ type: 'after' })
 @Service()
 export class ErrorMiddleware implements ExpressErrorMiddlewareInterface {
-  error(
-    error: ErrorInterface,
+  public error(
+    error: any,
     _req: Request,
     res: Response,
-    _next: (err?: ErrorInterface) => ErrorInterface
+    _next: (err?: any) => any
   ) {
-    if (DEV_ENV) {
-      console.error(`Message: ${error.message}`);
-      console.error(`Stack: ${error.stack}`);
+    const responseObject = {} as ErrorInterface;
+
+    // If the error comes from the class-validator then it's an array of ValidationError
+    if (
```

Comment:
maybe we could move this to a separate middleware to transform the error in case it is actually coming from class-validator, and letting this only handle the return. If not we could at least move the transformation to a private method here, wdyt?

Reply:
Yes for now I will create a private method, and then If we found thats necessary to move it to a different middleware, we go ahead and we do it.